tests/0081-admin.c

```diff
@@ -1183,6 +1596,14 @@ static void do_test_apis (rd_kafka_type_t cltype) {
         /* DescribeConfigs */
         do_test_DescribeConfigs(rk, mainq);
 
+        /* Delete records */
+        do_test_DeleteRecords("temp queue, op timeout 0", rk, NULL, 0);
+        do_test_DeleteRecords("main queue, op timeout 1500", rk, mainq, 1500);
+
+        /* Delete groups */
+        do_test_DeleteGroups("temp queue, op timeout 0", rk, NULL, 0);
+        do_test_DeleteGroups("main queue, op timeout 1500", rk, mainq, 1500);
+
```

Comment:
Please also add some dry API testing to 0080, like passing empty or NULL lists, test timeouts, etc, look at the other APIs for inspiration.

Reply:
Aha, testing the timeout path revealed that `rd_kafka_admin_worker` circumvents the fanout logic that the `*Response_parse` functions implement :)

Each fanned-out request that times out enqueues its own timeout error on the replyq, will have to do some extra work to consolidate them into a single response.

Reply:
I think the best way to ensure that a single response is always returned on the replyq is to do what I had originally intended and introduce a new type of ADMIN_FANOUT op with its own state machine similar to `rd_kafka_admin_worker`. The idea would be that if we want to fanout a request into N chunks, we create N+1 ops, the N chunks plus one ADMIN_FANOUT op that waits for and accumulates the responses from the chunks. 

The chunks operate like any other admin op and can succeed or fail due to broker errors, timeouts, etc. But instead of sending their responses to the client, they send them to a new queue that the ADMIN_FANOUT op owns.

The ADMIN_FANOUT op listens for response chunks on its queue, assembles them into a single response, and once it has received them all it sends the response back to the client. It can also be responsible for reordering the responses to match the order of the requests and propagating errors from chunks that failed outright (eg due to a timeout) to the specific elements of the request (ie if I try to delete 10 groups, and a chunk of 5 times out, the client response should always contain all 10 groups, with 5 marked as timed out). Note that since each chunk has its own timeout, the ADMIN_FANOUT op doesn't need to worry about timing out itself.

This is looking like a larger change though, so I wanted to run the design by you before getting too far into it. I also have some questions about the finer details.

1. I'm a little fuzzy on how ops are triggered from external sources like broker responses. Ideally, the ADMIN_FANOUT op would be triggered only when one of the chunks enqueues its response, because otherwise it can't have any work to do. But I'm not sure how to arrange that.

2. Even though chunks can fail individually for different reasons, the client response still needs an encompassing error code. How would you go about setting that? Just document that it will always be `NO_ERROR` and that the individual elements need to be checked?